Gemfile

```diff
@@ -15,8 +15,14 @@ gem 'rake'
 group :system_tests do
   beaker_version = ENV.fetch('BEAKER_VERSION', ['>= 4.17.0', '< 5.0.0'])
 
-  if "#{beaker_version}".include?('/')
-    gem 'beaker', :source => beaker_version
+  if "#{beaker_version}".include?(':')
+    # Just pass in BEAKER_VERSION as a string that would represent the usual
+    # hash of options.
+    #
+    # Something like: BEAKER_VERSION=':git => "https://my.repo/beaker.git" :tag => "1.2.3"'
+    #
+    # No, this isn't robust, but it's not really an 'every day' sort of thing
+    gem 'beaker', Hash[string.scan(/(\w+):\s+([^;]+)/).map { |k,v| [k.to_sym, v.strip] }]
```

Comment:
Wow.  I was just thinking of using something like `BEAKER_VERSION`, `BEAKER_GIT_URL`, and `BEAKER_GIT_REF`.

Reply:
@op-ct Yeah, I thought about that but then it got into "but what if I want to point at a directory, blah blah blah"

Reply:
But a case like that is... pretty plausible.  :thinking: 

`BEAKER_VERSION` seems too specific for a scope that broad.  How would you feel about `BEAKER_GEM`?

Reply:
`BEAKER_GEM_OPTIONS`?